pages/index.tsx

```diff
@@ -78,26 +78,18 @@ export default function Home({
       setNewGiftName('');
       setNewReceiver('');
     } catch (e) {
-      handleError(e);
+      toast(handleError(e), { type: 'error' });
     }
   }
 
   async function refreshGiftList() {
     try {
       setGiftData(await getAllGifts());
     } catch (e) {
-      handleError(e);
+      toast(handleError(e), { type: 'error' });
     }
   }
 
-  function handleError(e: unknown) {
-    const errorMessage = handleGeneralError(e);
-    if (errorMessage === 'You are unauthorized!') {
-      window.location.href = '/';
-    }
-    toast(errorMessage, { type: 'error' });
-  }
-
```

Comment:
Tämän poiston yhteydessä käyttäjää ei kirjata "pääsivulta" automaattisesti, jos hänen istunto on loppunut. Tulee pelkkä seuraava ilmoitus `Istuntosi on vanhentunut! Ole hyvä ja kirjaudu uudelleen jatkaaksesi!`

Olisikohan se käyttäjäystävällisempää, jos käyttäjä kirjattaisiin automaattisesti ulos ja siirrettäisiin kirjautumissivulle? :thinking: 

Jonkun vähän vastaavan funktion voisi tehdä, jolla tarkistetaan virhe ja kirjataan ulos tarvittaessa. Jotain vastaavaa:

```ts
function handleInternalError(e: unknown) {
    const errorMessage = handleError(e);
    if (errorMessage === 'You are unauthorized!') {
      // window.location.href:iä ei kannata käyttää, koska poistaa samalla Toastit
     // NextJS:n useRouter ei poista Toasteja
      window.location.href = '/';
    }
    toast(errorMessage, { type: 'error' });
  }
```

Reply:
> Olisikohan se käyttäjäystävällisempää, jos käyttäjä kirjattaisiin automaattisesti ulos ja siirrettäisiin kirjautumissivulle?

Kyllä olisi. Tuo olisi ehkä järkevin tehdä request-tasolla, niin ei tarvitse muistaa lisätä sitä kaikkiin paikkoihin. Eli esim. käyttää axion interceptoreita niin, että aina jos palvelin palauttaa 401 virheen niin ohjataan login-sivulle (tässä esim [SO kysymys aiheesta](https://stackoverflow.com/questions/47216452/how-to-handle-401-authentication-error-in-axios-and-react)). Routerin käyttö tosin ei taida onnistua axioksen interceptorissa 🤔

Reply:
https://github.com/samuliasmala/lahjalista/pull/45/files#r1817895751

Tässä tähän liittyen toteutus. Kommentissa on paremmin selitetty